### PR TITLE
Misc 1.16 changes

### LINKED ISSUE #1278: Docs generator ignores `@ZenCodeType.Field#value`
#### Issue Description:  
The docs generator uses the internal field name instead of the script-facing one, around [here?](https://github.com/CraftTweaker/CraftTweaker/blob/1.16/Crafttweaker_Annotation_Processors/src/main/java/com/blamejared/crafttweaker_annotation_processors/processors/document/conversion/converter/member/static_member/StaticFieldConverter.java#L36)

#### What happens:  
```java
	/** The manager for the standard Orechid. */
	@ZenCodeType.Field("main")
	public static final OrechidManager MAIN = new OrechidManager(() -> BotaniaAPIImpl.weights, "Orechid");
	/** The manager for the nether variant, Orechid Ignem. */
	@ZenCodeType.Field("nether")
	public static final OrechidManager NETHER = new OrechidManager(() -> BotaniaAPIImpl.netherWeights, "Orechid Ignem");
```
gets turned into
```markdown
## Static Properties

| Name | Type | Has Getter | Has Setter | Description |
|------|------|------------|------------|-------------|
| MAIN | [Orechid](/mods/Botania/Orechid) | true | false | The manager for the standard Orechid. |
| NETHER | [Orechid](/mods/Botania/Orechid) | true | false | The manager for the nether variant, Orechid Ignem. |
```

#### What you expected to happen:  
```markdown
## Static Properties

| Name | Type | Has Getter | Has Setter | Description |
|------|------|------------|------------|-------------|
| main | [Orechid](/mods/Botania/Orechid) | true | false | The manager for the standard Orechid. |
| nether | [Orechid](/mods/Botania/Orechid) | true | false | The manager for the nether variant, Orechid Ignem. |
```

#### crafttweaker.log file: n/a
____
#### Environment:  

- Minecraft Version: 1.16.5
- Forge Version: 36.0.46
- CraftTweaker Version: 7.1.0.291 (current latest)
- Are you using a server: n/a (annotation processor issue)

### LINKED ISSUE #1291: IItemStack.clearTag();
## Feature request name
Method to remove NBT tags from itemstack

## Feature request description
Add `IItemStack.withoutTag()` or `IItemStack.clearTag()` or allow null values in `IItemStack.withTag(@Nullable tag)` to remove NBT tag from itemstack

## Feature request reason
Might be useful for recipe transformers & event handlers. For me it's useful because we want to remove empty NBT tags from items but I think there are more cases when it will be useful

## Game Version
1.12+

### PR DESCRIPTION
**Pull request by friendlyhj**
Close #1291
Close #1278

### CODE REVIEW

**Comment by friendlyhj** [comment]
ready for review.

File: src/main/java/com/blamejared/crafttweaker/api/item/IItemStack.java [hunk 1]
```diff
@@ -412,6 +414,13 @@ default String getTranslationKey() {
      */
     @ZenCodeType.Method
     IItemStack withTag(IData tag);
+
+    /**
+     * Sets this ItemStack without a tag
```
Comment (jaredlll08) [inline src/main/java/com/blamejared/crafttweaker/api/item/IItemStack.java]:
```suggestion
     * Removes the tag from this ItemStack.
```

File: src/main/java/com/blamejared/crafttweaker/api/item/IItemStack.java [hunk 2]
```diff
@@ -630,6 +639,19 @@ default Item getDefinition() {
...
         return getInternal().getToolTypes().toArray(new ToolType[0]);
     }
+
+    @ZenCodeType.Getter("enchantments")
+    default Map<Enchantment, Integer> getEnchantments() {
+        return EnchantmentHelper.getEnchantments(getInternal());
+    }
+
+    /**
+     * Gets the level of given enchantment on the item. Returns 0 if the item doesn't have given enchantment.
```
Comment (jaredlll08) [inline src/main/java/com/blamejared/crafttweaker/api/item/IItemStack.java]:
```suggestion
     * Gets the level of the given enchantment on the item. Returns 0 if the item doesn't have the given enchantment.
```

File: src/main/java/com/blamejared/crafttweaker/impl_native/loot/ExpandLootContext.java [hunk 3]
```diff
@@ -168,14 +170,13 @@ public static int getLootingModifier(final LootContext internal) {
     }
 
     /**
-     * Gets the world where the interaction happened, if it exists or can be obtained; null otherwise.
+     * Gets the world where the interaction happened.
```
Comment (jaredlll08) [inline src/main/java/com/blamejared/crafttweaker/impl_native/loot/ExpandLootContext.java:173]: So this looks fine to me, but if I remember correctly, @TheSilkMiner has said that the world sometimes doesn't exist, so just going to ping him so he knows the doc is changing.

I took a look at the code, a world will always exist as it is required to get the loot table manager.

**Review by jaredlll08 — APPROVED** [review]

**Pull request merged**